src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -218,6 +218,17 @@ public void WriteBlob(BinaryLogRecordKind kind, byte[] bytes)
             Write(bytes);
         }
 
+        public void WriteBlob(BinaryLogRecordKind kind, Stream stream)
+        {
+            // write the blob directly to the underlying writer,
+            // bypassing the memory stream
+            using var redirection = RedirectWritesToOriginalWriter();
+
+            Write(kind);
+            Write(stream.Length);
```

Comment:
Note that `stream.Length` is `long` while the original code was writing down `int`. These are serialized differently and thus this breaks the binary format:
```csharp
        private void Write(int value)
        {
            BinaryWriterExtensions.Write7BitEncodedInt(binaryWriter, value);
        }

        private void Write(long value)
        {
            binaryWriter.Write(value);
        }
```

Reply:
Original `BinaryWriter` also utilizes a lot of overloads. Your design just follows that, but it's easy to miss this type of bug.